scratch/misc/percent_flagged.R

```diff
@@ -0,0 +1,82 @@
+generate_flag_report <- function(df, df_index) {
+
+  # Extract the site and parameter from the df_name
+  site <- sub("^([^\\-]+)-.*$", "\\1", df_index)
+  parameter <- sub("^[^-]+-(.*)$", "\\1", df_index)
+
+  list_of_flags <- c("sonde not employed",
+                     "site visit",
+                     "sv window",
+                     "sensor specification range",
+                     "seasonal range",
+                     "slope flag suspect",
+                     "outside sd range",
+                     "repeated value",
+                     "missing data",
+                     "24hr anomaly",
+                     "anomaly window")
+
+  # summarize total data points
+  total_observations <- df %>%
+    summarise(n_total = n_distinct(DT_round)) %>%
+    pull(n_total)
+
+  # summarize total days
+  total_observations_dates <- df %>%
+    group_by(date = format(DT_round, "%m-%d-%Y")) %>%
+    summarize(n_total = nrow(date)) %>%
+    nrow()
+
+  row_list <- list()
+  for (i in list_of_flags) {
```

Comment:
I challenge you to functionalize this! :)

Reply:
you could either do that as a separate function that you call here 
OR! 
you can map and define the function:
`map(list_of_flags, function(x){` 
and then mostly replace `i` with `x` (and a few other tweaks, but I think you get where I'm going here!)